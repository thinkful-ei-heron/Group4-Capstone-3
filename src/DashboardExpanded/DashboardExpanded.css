.dashboard-expanded-section {
	width: 90vh;
	background-color: buttonface;
	color: Black;
	border-radius: 5%;
	margin-top: 1rem;
	margin-left: auto;
	margin-right: auto;
	height: fit-content;
	padding: 1rem 0 1.5rem 0;
}

.dashboard-expanded-section:hover {
	background-color: rgba(209, 240, 205, 0.884);
	cursor: pointer;
}

.edit-rating-img {
	margin: 1rem auto;
	display: block;
}

.edit-beer-form fieldset {
	width: 50%;
	margin-top: 20px;
	border-radius: 10px;
	border: 2px solid rgba(60, 131, 64, 0.858);
	padding: 20px;
	box-shadow: inset 0px 0px 15px rgb(243, 230, 230);
	-moz-box-shadow: inset 0px 0px 15px rgb(236, 217, 217);
	-webkit-box-shadow: inset 0px 0px 15px rgb(215, 209, 209);
	background-color: rgba(255, 255, 255, 0.66);
	z-index: 1;
}

.edit-beer-form legend {
	color: rgb(21, 53, 30);
	border-top: 2px solid rgb(73, 159, 70);
	border-left: 2px solid rgb(73, 159, 70);
	border-right: 2px solid rgb(95, 155, 100);
	border-radius: 5px 5px 0px 0px;
	-webkit-border-radius: 5px 5px 0px 0px;
	-moz-border-radius: 5px 5px 0px 0px;
	background: rgba(226, 240, 227, 0.653);
	padding: 0px 8px 3px 8px;
	box-shadow: -0px -1px 2px #f1f1f1;
	-moz-box-shadow: -0px -1px 2px #f1f1f1;
	-webkit-box-shadow: -0px -1px 2px #f1f1f1;
	font-weight: normal;
	font-size: 20px;
}

.edit-buttons {
	margin-top: 15%;
}
