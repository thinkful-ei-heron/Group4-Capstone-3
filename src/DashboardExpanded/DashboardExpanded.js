import React from 'react';
import UserContext from '../contexts/UserContext';
import Utils from '../Utils/Utils';
import ReactTooltip from 'react-tooltip';
import './DashboardExpanded.css';

class DashboardExpanded extends React.Component {
	static contextType = UserContext;

	constructor(props) {
		super(props);
		this.state = {
			journal: this.props.journal,
			isEditable: false,
			name: '',
			date_created: '',
			location: '',
			description: '',
			type: '',
			rating: 0,
			abv: 0,
			heaviness: 0,
			color: 0
		};
	}
	componentDidUpdate(prevProps, prevState, snapshot) {
		if (prevProps !== this.props) {
			this.setState({
				name: this.props.journal.name,
				date_created: this.props.journal.date_created,
				location: this.props.journal.location,
				description: this.props.journal.description,
				type: this.props.journal.type,
				rating: this.props.journal.rating,
				abv: this.props.journal.abv,
				heaviness: this.props.journal.heaviness,
				color: this.props.journal.color
			});
			this.forceUpdate();
		}
	}
	handleName = (e) => {
		this.setState({ name: e.target.value });
	};
	handleDate = (e) => {
		this.setState({ date_created: e.target.value });
	};
	handleLoc = (e) => {
		this.setState({ location: e.target.value });
	};
	handleDesc = (e) => {
		this.setState({ description: e.target.value });
	};
	handleType = (e) => {
		this.setState({ type: e.target.value });
	};
	handleRating = (e) => {
		this.setState({ rating: e.target.value });
	};
	handleAbv = (e) => {
		this.setState({ abv: e.target.value });
	};
	handleHeaviness = (e) => {
		this.setState({ heaviness: e.target.value });
	};
	handleColor = (e) => {
		this.setState({ color: e.target.value });
	};

	toggleIsEditable = () => {
		this.setState({
			isEditable: !this.state.isEditable,
			name: this.props.journal.name,
			date_created: this.props.journal.date_created,
			location: this.props.journal.location,
			description: this.props.journal.description,
			type: this.props.journal.type,
			rating: this.props.journal.rating,
			abv: this.props.journal.abv,
			heaviness: this.props.journal.heaviness,
			color: this.props.journal.color
		});
	};

	render() {
		let x = this.state.isEditable ? (
			<form className ='edit-beer-form'
				onSubmit={() =>
					this.props.handleSubmit(this.props.journal.id, {
						id: this.props.journal.id,
						name: this.state.name,
						date_created: this.state.date_created,
						location: this.state.location,
						description: this.state.description,
						type: this.state.type,
						rating: this.state.rating,
						abv: parseFloat(this.state.abv),
						heaviness: parseInt(this.state.heaviness),
						color: parseInt(this.state.color)
					})}>
				<section className='form-section'>
					<fieldset>
						<legend>Beer Information</legend>
						<figure>
							<img src={Utils.getImage(parseInt(this.state.color))} alt={'Beer'} width={'150px'} />
						</figure>
						<div>
							<label>
								Date: <input type="date" value={this.state.date_created} onChange={this.handleDate} />
							</label>
						</div>
						<div>
							<label>
								Location: <input type="text" value={this.state.location} onChange={this.handleLoc} />
							</label>
						</div>
						<div>
							<label>
									Name: <input type="text" value={this.state.name} onChange={this.handleName} />
							</label>
						</div>
						<div>
						<label>
								Type: 
								<select value={this.state.type} onChange={this.handleType}>
									<option value='Ale'>Ale</option>
									<option value='Altbier'>Altbier</option>
									<option value='American Lager'>American Lager</option>
									<option value='Barley Wine'>Barley Wine</option>
									<option value='Belgian'>Belgian</option>
									<option value='Berliner Weisse'>Berliner Weisse</option>
									<option value='Bitter'>Bitter</option>
									<option value='Bock'>Bock</option>
									<option value='Brown Ale'>Brown Ale</option>
									<option value='Cider'>Cider</option>
									<option value='Cream Ale'>Cream Ale</option>
									<option value='Doppelbock'>Doppelbock</option>
									<option value='Dunkel'>Dunkel</option>
									<option value='Flanders Red Ale'>Flanders Red Ale</option>
									<option value='German Pilser'>German Pilser</option>
									<option value='Gose'>Gose</option>
									<option value='Helles'>Helles</option>
									<option value='Helles Bock'>Helles Bock</option>
									<option value='Honey'>Honey</option>
									<option value='Imperial IPA'>Imperial IPA</option>
									<option value='IPA'>IPA</option>
									<option value='Irish Red Ale'>Irish Red Ale</option>
									<option value='Kolsch'>Kolsch</option>
									<option value='Lager'>Lager</option>
									<option value='Lambic'>Lambic</option>
									<option value='Mild ale'>Mild ale</option>
									<option value='Old Ale'>Old Ale</option>
									<option value='Pale Ale'>Pale Ale</option>
									<option value='Pale Lager'>Pale Lager</option>
									<option value='Pilsner'>Pilsner</option>
									<option value='Porter'>Porter</option>
									<option value='Quadrupel'>Quadrupel</option>
									<option value='Rye'>Rye</option>
									<option value='Saison'>Saison</option>
									<option value='Schwarzbier'>Schwarzbier</option>
									<option value='Scotch Ale'>Scotch Ale</option>
									<option value='Seasonal Beer'>Seasonal Beer</option>
									<option value='Stout'>Stout</option>
									<option value='Vienna lager'>Vienna lager</option>
									<option value='Wittbier'>Wittbier</option>
									<option value='Other'>Other</option>
								</select>
							</label>
						</div>
						<div>
							<label data-tip='Alcohol By Volume'>
							<ReactTooltip place="top" type="dark" effect="solid"/>
								ABV: <input type="number" value={this.state.abv} onChange={this.handleAbv} /> %
							</label>
						</div>
						<div>
							<h3 data-tip='Color of beer'>Dark to Light</h3>
							<input type="range" min="1" max="6" value={this.state.color} onChange={this.handleColor} />
						</div>
						<div>
							<h3 data-tip='Strength of alcohol'>Heavy to Light</h3>
							<input type="range" min="1" max="5" value={this.state.heaviness} onChange={this.handleHeaviness} />
						</div>
					</fieldset>
					<fieldset className='lastfield'>
						<legend>What I think about this beer!</legend>
						<div>
							<figure>
								<figcaption>Rating</figcaption>
								<img className='edit-rating-img' src={Utils.getRatingImage(parseInt(this.state.rating))} alt={'Rating icons'} height={'50px'} />
								<input type="range" min="0" max="9" value={this.state.rating} onChange={this.handleRating} />
							</figure>
						</div>
						<div>
							<label>
								Description <textarea value={this.state.description} onChange={this.handleDesc} />
							</label>
						</div>
						<div className="edit-buttons">
							<button className="clear-submit-back-btn" onClick={this.toggleIsEditable}>Cancel</button>
							<button className="clear-submit-back-btn"type={'submit'}>Submit</button>
						</div>
					</fieldset>
				</section>
			</form>
		) : (
            <>
                <div className='dashboard-expanded-section'>
                <section onClick={() => this.props.toggleExpanded(this.state.journal.id)}>
                    <figure>
                        <img className='beer-color' src={Utils.getImage(this.state.journal.color)} alt={'Beer'} width={'150px'} />
                        <figcaption>ABV: {this.state.journal.abv} % </figcaption>
                    </figure>
                    <section>
												<h3 className='expanded-beer-name'>
                            <span>{this.state.journal.name}</span>
                        </h3>
												<div className='expanded-small-details'>
													<h3 className='expanded-location'>
															<span>{this.state.journal.location}</span>
													</h3>
													<h3 className='expanded-type'>
															<span>{this.state.journal.type}</span>
													</h3>
												</div>
                    </section>
                    {/* <input type="range" min="0" max="9" value={this.state.journal.rating} readOnly /> */}
										<fieldset className='expanded-description'>
											<legend>
												<span className='expanded-date'>{Utils.formattedDate(this.state.journal.date_created)}</span>
											</legend>
											{this.state.journal.description}
										</fieldset>
										<div>
											<h3 className='expanded-slide-label'>Color</h3>
										</div>
										<section className='expanded-sliders-section'>
											<p>Light</p>
											<input type="range" min="1" max="6" value={this.state.journal.color} readOnly className='expanded-slider'/>
											<p>Dark</p>
										</section>
										<div>
											<h3 className='expanded-slide-label'>Flavor</h3>
										</div>
										<section className='expanded-sliders-section'>
											<p>Light</p>
											<input type="range" min="1" max="5" value={this.state.journal.heaviness} readOnly className='expanded-slider'/>
											<p>Heavy</p>
										</section>
										<figure>
											<img className='rating-img' src={Utils.getRatingImage(this.state.journal.rating)} alt={'Rating icons'} height={'50px'} />
										</figure>
                </section>
								<button className='clear-submit-back-btn'onClick={this.toggleIsEditable}>Edit</button>
                <button className='clear-submit-back-btn'onClick={() => this.props.handleDelete(this.state.journal.id)}>Delete</button>
                </div>
            </>
		);
		return <li>{x}</li>;
	}
}

export default DashboardExpanded;
