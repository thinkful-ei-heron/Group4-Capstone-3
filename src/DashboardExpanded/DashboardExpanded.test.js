import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { BrowserRouter } from 'react-router-dom';
import DashboardExpanded from './DashboardExpanded'

Enzyme.configure({ adapter: new Adapter() }); 

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <BrowserRouter>
      <DashboardExpanded journal={'color'}/>
    </BrowserRouter>,
    div)
  ReactDOM.unmountComponentAtNode(div)
})
it('renders the complete page', () => {
  const wrapper = shallow (<DashboardExpanded journal={'color'}/>)
  expect(toJson(wrapper)).toMatchSnapshot()
})

it('renders the DashboardExpanded given props', () => {
  const wrapper = shallow(<DashboardExpanded journal={'color'} />)
  expect(toJson(wrapper)).toMatchSnapshot()
})