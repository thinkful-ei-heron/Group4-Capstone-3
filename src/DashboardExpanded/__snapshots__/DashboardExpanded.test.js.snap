// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the DashboardExpanded given props 1`] = `
<li>
  <div
    className="dashboard-expanded-section"
  >
    <section
      onClick={[Function]}
    >
      <figure>
        <img
          alt="Beer"
          className="beer-color"
          src="light.jpg"
          width="150px"
        />
        <figcaption>
          ABV: 
           % 
        </figcaption>
      </figure>
      <section>
        <h3
          className="expanded-beer-name"
        >
          <span />
        </h3>
        <div
          className="expanded-small-details"
        >
          <h3
            className="expanded-location"
          >
            <span />
          </h3>
          <h3
            className="expanded-type"
          >
            <span />
          </h3>
        </div>
      </section>
      <fieldset
        className="expanded-description"
      >
        <legend>
          <span
            className="expanded-date"
          >
            NaN/NaN/NaN
          </span>
        </legend>
      </fieldset>
      <div>
        <h3
          className="expanded-slide-label"
        >
          Color
        </h3>
      </div>
      <section
        className="expanded-sliders-section"
      >
        <p>
          Dark
        </p>
        <input
          className="expanded-slider"
          max="6"
          min="1"
          readOnly={true}
          type="range"
        />
        <p>
          Light
        </p>
      </section>
      <div>
        <h3
          className="expanded-slide-label"
        >
          Flavor
        </h3>
      </div>
      <section
        className="expanded-sliders-section"
      >
        <p>
          Heavy
        </p>
        <input
          className="expanded-slider"
          max="5"
          min="1"
          readOnly={true}
          type="range"
        />
        <p>
          Light
        </p>
      </section>
      <figure>
        <img
          alt="Rating icons"
          className="rating-img"
          height="50px"
          src="half.png"
        />
      </figure>
    </section>
    <button
      className="clear-submit-back-btn"
      onClick={[Function]}
    >
      Edit
    </button>
    <button
      className="clear-submit-back-btn"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</li>
`;

exports[`renders the complete page 1`] = `
<li>
  <div
    className="dashboard-expanded-section"
  >
    <section
      onClick={[Function]}
    >
      <figure>
        <img
          alt="Beer"
          className="beer-color"
          src="light.jpg"
          width="150px"
        />
        <figcaption>
          ABV: 
           % 
        </figcaption>
      </figure>
      <section>
        <h3
          className="expanded-beer-name"
        >
          <span />
        </h3>
        <div
          className="expanded-small-details"
        >
          <h3
            className="expanded-location"
          >
            <span />
          </h3>
          <h3
            className="expanded-type"
          >
            <span />
          </h3>
        </div>
      </section>
      <fieldset
        className="expanded-description"
      >
        <legend>
          <span
            className="expanded-date"
          >
            NaN/NaN/NaN
          </span>
        </legend>
      </fieldset>
      <div>
        <h3
          className="expanded-slide-label"
        >
          Color
        </h3>
      </div>
      <section
        className="expanded-sliders-section"
      >
        <p>
          Dark
        </p>
        <input
          className="expanded-slider"
          max="6"
          min="1"
          readOnly={true}
          type="range"
        />
        <p>
          Light
        </p>
      </section>
      <div>
        <h3
          className="expanded-slide-label"
        >
          Flavor
        </h3>
      </div>
      <section
        className="expanded-sliders-section"
      >
        <p>
          Heavy
        </p>
        <input
          className="expanded-slider"
          max="5"
          min="1"
          readOnly={true}
          type="range"
        />
        <p>
          Light
        </p>
      </section>
      <figure>
        <img
          alt="Rating icons"
          className="rating-img"
          height="50px"
          src="half.png"
        />
      </figure>
    </section>
    <button
      className="clear-submit-back-btn"
      onClick={[Function]}
    >
      Edit
    </button>
    <button
      className="clear-submit-back-btn"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</li>
`;
